src/status.ts

```diff
@@ -23,123 +23,113 @@ const STARTED_SERVER_OPTIONS = [
   { label: "Ruby LSP: Restart", description: Command.Restart },
 ];
 
-export class StatusItem {
-  private context: vscode.ExtensionContext;
-  private ruby: Ruby;
-  private selector: vscode.DocumentSelector;
-  private serverStatus: vscode.LanguageStatusItem;
-  private yjitStatus: vscode.LanguageStatusItem;
-  private rubyVersionStatus: vscode.LanguageStatusItem;
-  private experimentalFeaturesStatus: vscode.LanguageStatusItem;
+abstract class Status {
+  public item: vscode.LanguageStatusItem;
+  protected context: vscode.ExtensionContext;
 
-  constructor(context: vscode.ExtensionContext, ruby: Ruby) {
-    this.context = context;
-    this.ruby = ruby;
-    this.selector = {
+  constructor(id: string, context: vscode.ExtensionContext, _ruby: Ruby) {
```

Comment:
It feels like a better API might be:
```suggestion
  constructor(id: string, client: { context: vscode.ExtensionContext, ruby: Ruby }) {
```
so that status items store references to the `client` and when `refresh` is called, no arguments need to be passed, they can each access the relevant value from the `client`.

Since the typing for `client` is structural, you can easily replace that in tests with a fake client instance that has the same two properties.

Reply:
I like this! My original idea was to pass the actual client object to have easy access to the state. But unfortunately, that introduces a circular dependency and TypeScript won't have it.